### PR TITLE
Enable console logging if MONO or LOGGER is defined

### PR DESCRIPTION
**Pull request by jcoby**
Re-enable console logging on linux and mac builds with the -DMONO option.

Rename overloaded Debug_ConsolePrintf function to
Debug_ConsolePrintfAt to fix macro expansion.

Adds -DLOGGER alias for -DMONO.

Implements #157

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,13 +54,13 @@ endif()
...
   message("Building for Linux")
-  add_definitions( -D_DEBUG -D__LINUX__ -DLINUX -D_MAX_PATH=260 -D_MAX_FNAME=256 -D_REENRANT -D__32BIT__ -DHAVEALLOCA_H -D_USE_OGL_ACTIVE_TEXTURES)
+  add_definitions( -D_DEBUG -D__LINUX__ -DLINUX -D_MAX_PATH=260 -D_MAX_FNAME=256 -D_REENRANT -D__32BIT__ -DHAVEALLOCA_H -D_USE_OGL_ACTIVE_TEXTURES -DLOGGER)
   set(PLATFORM_INCLUDES "lib/linux" ${SDL_INCLUDE_DIR})
 endif()
 
 if(APPLE)
   message("Building for MAC OSX")
-  add_definitions(-D_DEBUG -D__LINUX__ -DLINUX -D_MAX_PATH=260 -D_MAX_FNAME=256 -D_REENRANT -DMACOSX=1 -D_USE_OGL_ACTIVE_TEXTURES)
+  add_definitions(-D_DEBUG -D__LINUX__ -DLINUX -D_MAX_PATH=260 -D_MAX_FNAME=256 -D_REENRANT -DMACOSX=1 -D_USE_OGL_ACTIVE_TEXTURES -DLOGGER)
```
Comment (Lgt2x) [inline CMakeLists.txt:63]: we want this definition conditionally, if the user specified the CMake [Option](https://cmake.org/cmake/help/latest/command/option.html) LOGGER

File: lib/mono.h [hunk 2]
```diff
@@ -66,12 +66,12 @@
 #include "debug.h"
 void nw_InitTCPLogging(char *ip, unsigned short port);
 void nw_TCPPrintf(int n, char *format, ...);
-#if (!defined(RELEASE)) && defined(MONO)
+#if (!defined(RELEASE)) && (defined(MONO) || defined(LOGGER))
```
Comment (Lgt2x) [inline lib/mono.h:69]: I don't think we need to keep both, `LOGGER` is enough

Comment (jcoby) [reply lib/mono.h]: I wanted to keep both so we could keep the MONO code for historical reasons if we add a logging library. But it's easy enough to resurrect MONO later.

File: CMakeLists.txt (hunk 1 above)
Comment (jcoby) [reply CMakeLists.txt]: Added in latest commit

**Comment by Lgt2x** [comment]
Tested, works great. Thanks!

**Pull request merged**